apps/src/lab2/views/LabViewsRenderer.tsx

```diff
@@ -38,27 +41,60 @@ const LabViewsRenderer: React.FunctionComponent = () => {
   );
 
   const isViewingExemplar = getAppOptionsViewingExemplar();
-
-  const capitalizedLessonBackground = useAppSelector(
-    state =>
-      capitalizeFirstLetter(
-        getCurrentLesson(state)?.background || 'light'
-      ) as Theme
+  const lesson = useAppSelector(state => getCurrentLesson(state));
+  const {signInState} = useAppSelector(state => state.currentUser);
+  const dispatch = useAppDispatch();
+
+  // We only use the global user preference for theme if the current lesson has
+  // at least one python lab level or the current level is a python lab level.
+  const useThemeUserPreference = useMemo(
+    () =>
+      levelProperties?.appName === 'pythonlab' ||
+      lesson?.levels.some((level: Level) => level.app === 'pythonlab'),
+    [lesson?.levels, levelProperties?.appName]
   );
 
   // Set the theme for the current app.
   const {setTheme} = useTheme();
   useEffect(() => {
     if (currentAppName) {
       const supportedThemes = lab2EntryPoints[currentAppName]?.themes;
-
-      if (supportedThemes.includes(capitalizedLessonBackground)) {
-        setTheme(capitalizedLessonBackground);
+      dispatch(setIsLoadingTheme(true));
+
+      const setThemeHelper = () => {
+        // If the body has a class use that to set the theme, if its supported by the lab.
+        // Otherwise, use the first supported theme.
+        // We will only run this if we are not using the user preference, so it's safe to pull
+        // from the body class, as the user is not dynamically changing the theme.
```

Comment:
When changing levels, looks like we don't take the new level's `background` into account if there's no lesson background, right? I don't think we do today either, but just wondering if we want to consider that (or if we'd like to phase out level-specific backgrounds altogether).

Reply:
Yeah we weren't ever doing anything with level background in lab2 currently. I think we probably don't really want to use level background (or I would argue lesson background?) much going forward.